druid-shell/src/mac/mod.rs

```diff
@@ -235,14 +271,39 @@ extern "C" fn draw_rect(this: &mut Object, _: Sel, dirtyRect: NSRect) {
         let view_state: *mut c_void = *this.get_ivar("viewState");
         let view_state = &mut *(view_state as *mut ViewState);
         let anim = (*view_state).handler.paint(&mut piet_ctx);
+        piet_ctx.finish();
+        // TODO: log errors
 
-        // TODO: handle animation
+        if anim {
+            // TODO: synchronize with screen refresh rate using CVDisplayLink instead.
+            let () = msg_send!(this as *const _, performSelectorOnMainThread: sel!(redraw)
+                withObject: nil waitUntilDone: NO);
+        }
 
         let superclass = msg_send![this, superclass];
         let () = msg_send![super(this, superclass), drawRect: dirtyRect];
     }
 }
 
+extern "C" fn run_idle(this: &mut Object, _: Sel) {
+    unsafe {
```

Comment:
Looks like you might be wrapping more code in unsafe than is strictly necessary. I think it's best to only wrap the absolute minimum in unsafe as possible.

Reply:
This is a style point I don't have strong feelings about, but have tightened up the scope of unsafe in cases where it doesn't force gymnastics for visibility.